types/wordpress__components/flex/index.d.ts

```diff
@@ -0,0 +1,57 @@
+import { ComponentType, HTMLProps, CSSProperties } from 'react';
+
+declare namespace Flex {
+    type FlexAlignments =
+        | 'start'
+        | 'flex-start'
+        | 'flex-end'
+        | 'center'
+        | 'end'
+        | 'normal'
+        | 'baseline'
+        | 'first baseline'
+        | 'last baseline'
+        | 'stretch'
+        | 'safe'
+        | 'unsafe';
+
+
+    interface FlexProps {
+        /**
+         * Sets align-items. Top and bottom are shorthand for flex-start and flex-end respectively.
+         */
+        align?: 
+            | FlexAlignments
+            | 'top'
+            | 'bottom'
+            | 'self-start'
+            | 'self-end';
+
+        /**
+         * The size of padding to put between the children of the container.
+         */
+        gap?: number;
+
+        /**
+         * Sets jusifty-content. Left and right are shorthand for flex-start and flex-end respectively, not the actual CSS value.
+         */
+        justify?: 
+            | FlexAlignments
+            | 'left'
+            | 'right'
+            | 'space-between'
+            | 'space-around'
+            | 'space-evenly';
```

Comment:
Ah, I tried this at first but was seeing a complation error. Running `npm install` in the `react` type fixed it though (I was missing the `csstype` dependency being installed). Should I open a separate PR to declare `react` an explicit dependency of these types? Or do we expect it to come through from `@wordpress/element`? In which case I'm left confused why it's not 🤔

Reply:
I think it makes sense for this package to depend on React types directly.

Related: https://github.com/WordPress/gutenberg/pull/25086

When we made `@wordpress/element` provide its own types, we had to decide whether to re-export all of the React helper types (which is what the DT types did) or not. We chose not to re-export the React types from `@wordpress/element`, some of that decision is documented in this issue: https://github.com/WordPress/gutenberg/issues/21767:

> While TypeScript has clear ways of importing and exporting types, I was unable to re-export React types from `@wordpress/element` via JSDoc. At the very least, this would mean taking on a significant maintenance burden.